operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -354,8 +355,14 @@ func CreateOrUpdateLokiStack(
 	// be possibly misaligned with the configmap, which could lead to
 	// a user possibly being unable to read logs.
 	if err := status.SetStorageSchemaStatus(ctx, k, req, storageSchemas); err != nil {
-		ll.Error(err, "failed to set storage schema status")
-		return err
+		if errors.Is(err, status.WarningError) {
+			if er := status.SetWarningCondition(ctx, k, req, status.StorageSchemaNeedsUpgrade, lokiv1.ReasonObjectStorageSchemaShouldBeUpgraded); er != nil {
+				return kverrors.Wrap(er, "error setting warning condition")
+			}
+		} else {
+			ll.Error(err, "failed to set storage schema status")
+			return err
+		}
```

Comment:
I would simplify this moving SetWarningCondition call direct into `SetStorageSchemaStatus`.